src/compute/ICompute.java

```diff
@@ -12,5 +12,8 @@ public interface ICompute {
 	 * @return the resultant long
 	 */
 	long compute(int i);
+
+	boolean isxPrime(int x);
+	int firstgreaterprimeNum(int x, int y);
```

Comment:
```suggestion

```
These methods are only pertinent to our specific implementation of a Compute Engine. They should not be in the interface. Interfaces are only for methods that interacting classes should know about. In our case, the only method that a user of an `ICompute`-implementing object needs to know about is the `compute(int i)` method. A caller of this method trusts that the implementing class will do as it is told. It doesn't need to know about the inner workings of the class.